proposals/stdlib/hex-format.md

```diff
@@ -0,0 +1,320 @@
+  `hex` and `fromhex` functions on [bytes objects](https://docs.python.org/3/library/stdtypes.html#bytes-objects).
+
+## Proposal
+
+Considering the use cases mentioned above it is proposed to have the following format options.
+
+For formatting a numeric value:
+* Whether upper case or lower case hexadecimal digits should be used
+* The prefix of the hex representation
+* The suffix of the hex representation
+* Whether to remove leading zeros in the hex representation
+
+For formatting `ByteArray`:
+* Whether upper case or lower case hexadecimal digits should be used
+* The number of bytes per line
+* The number of bytes per group
+* The string used to separate groups in a line
+* The string used to separate bytes in a group
+* The prefix of a byte hex representation
+* The suffix of a byte hex representation
+
+### Creating a format
+
+It is proposed to introduce an immutable `HexFormat` class that holds the options.
+`Builder` is used to configure a format. Each option in the builder has a default value that can be customized.
+All related types are nested inside `HexFormat` to reduce top-level surface area of the API:
+```
+public class HexFormat internal constructor(
+    val upperCase: Boolean,
+    val bytes: BytesHexFormat,
+    val number: NumberHexFormat
+) {
+
+    public class Builder internal constructor() {
+        var upperCase: Boolean = false
+        val bytes: BytesHexFormat.Builder = BytesHexFormat.Builder()
+        val number: NumberHexFormat.Builder = NumberHexFormat.Builder()
+        
+        inline fun bytes(builderAction: BytesHexFormat.Builder.() -> Unit)
+        inline fun number(builderAction: NumberHexFormat.Builder.() -> Unit)
+    }
+  
+    public class BytesHexFormat internal constructor(
+        val bytesPerLine: Int,
+        val bytesPerGroup: Int,
+        val groupSeparator: String,
+        val byteSeparator: String,
+        val bytePrefix: String,
+        val byteSuffix: String
+    ) {
+
+        public class Builder internal constructor() {  
+            var bytesPerLine: Int = Int.MAX_VALUE  
+            var bytesPerGroup: Int = Int.MAX_VALUE  
+            var groupSeparator: String = "  "  
+            var byteSeparator: String = ""  
+            var bytePrefix: String = ""  
+            var byteSuffix: String = ""  
+        }
+    }  
+
+    public class NumberHexFormat internal constructor(
+        val prefix: String,
+        val suffix: String,
+        val removeLeadingZeros: Boolean
+    ) {
+
+        public class Builder internal constructor() {  
+            var prefix: String = ""  
+            var suffix: String = ""  
+            var removeLeadingZeros: Boolean = false  
+        }
+    }
+}
+```
+
+`BytesHexFormat` and `NumberHexFormat` classes hold format options for `ByteArray` and numeric values, correspondingly.
+`uppercase` option, which common to both `ByteArray` and numeric values, is stored in `HexFormat`.
```

Comment:
```suggestion
`uppercase` option, which is common to both `ByteArray` and numeric values, is stored in `HexFormat`.
```

Reply:
Hi,
I think the `uppercase` option is vague.
Does `uppercase = false` means force all letters to lowercase or does it mean keep the letters as is (not touch them)?
Can it be renamed to something like `forceUppercase` or similar?